pkgs/top-level/all-packages.nix

```diff
@@ -22450,6 +22450,10 @@ with pkgs;
 
   libarchive-qt = libsForQt5.callPackage ../development/libraries/libarchive-qt { };
 
+  libaribcaption = callPackage ../by-name/li/libaribcaption/package.nix {
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreGraphics CoreText;
+  };
```

Comment:
I hope you don't mind me asking, but I thought the whole purpose of `pkgs/by-name` was to avoid `all-packages.nix`, with the exception of [changing implicit defaults](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults). Is this commonplace for darwin packages?

Reply:
Yes, e.g.

```nix
    cargo-rdme = callPackage ../by-name/ca/cargo-rdme/package.nix {
      inherit (darwin.apple_sdk.frameworks) Security;
    };
```